File: client/src/pages/page/type/page.ts
import { PageComponents } from "@/pages/page/type/pageComponents";

export interface PageBase {
  id: number;
  shopId: number;
  title: string;
  previewImg: string;
  styleType: number; //页面样式
  mode: number; //页面归属
  isDelete: boolean;
}

export interface Page extends PageBase {
  components: (PageComponents | null)[];
}

export interface PageHttp extends PageBase {
  components: string | null;
}


export enum StyleType {
  default, self
}

export enum PageMode {
  default, self
}

File: service/app/model/Page.scala
package model

import java.util.Date

import play.api.libs.json._


case class Page(
                 id: Int,
                 shopId: Int,
                 title: String,
                 mode: Int,
                 styleType: Int,
                 previewImg: Option[String],
                 backgroundColor: Option[String],
                 components: Option[String],
                 isDelete: Option[Boolean],
                 createTime: Option[Date],
                 updateTime: Option[Date],
               )

object Page {
  implicit val pageFormat: Format[Page] = Json.format[Page]
}


